atlasdb-dbkvs-tests/src/test/java/com/palantir/atlasdb/keyvalue/dbkvs/impl/oracle/OracleAlterTableTest.java

```diff
@@ -0,0 +1,180 @@
+                    .build())
+            .build();
+
+    private static final DbKeyValueServiceConfig CONFIG_WITH_ALTER = ImmutableDbKeyValueServiceConfig.builder()
+            .from(CONFIG)
+            .ddl(ImmutableOracleDdlConfig.builder()
+                    .overflowMigrationState(OverflowMigrationState.FINISHED)
+                    .addAlterTablesOrMetadataToMatch(TABLE_REFERENCE)
+                    .build())
+            .build();
+
+    @ClassRule
+    public static final TestResourceManager TRM =
+            new TestResourceManager(() -> ConnectionManagerAwareDbKvs.create(CONFIG));
+
+    private static final String COLUMN_NAME = "variable";
+
+    private static final TableMetadata EXPECTED_TABLE_METADATA = TableMetadata.builder()
+            .singleRowComponent("name", ValueType.STRING)
+            .singleNamedColumn("var", COLUMN_NAME, ValueType.STRING)
+            .build();
+
+    private static final TableMetadata OLD_TABLE_METADATA = TableMetadata.builder()
+            .singleRowComponent("name", ValueType.STRING)
+            .singleNamedColumn("var", COLUMN_NAME, ValueType.FIXED_LONG)
+            .build();
+
+    private static final Cell DEFAULT_CELL = createCell("foo", "bar");
+    private static final Cell DEFAULT_CELL_2 = createCell("something", "bar");
+
+    private static final byte[] DEFAULT_VALUE = PtBytes.toBytes("amazing");
+    private static final long TIMESTAMP = 1L;
+
+    private ConnectionManagerAwareDbKvs defaultKvs;
+    private ConnectionSupplier connectionSupplier;
+    private OracleTableNameGetter oracleTableNameGetter;
+
+    @Before
+    public void setup() {
+        defaultKvs = (ConnectionManagerAwareDbKvs) TRM.getDefaultKvs();
+        connectionSupplier = DbKvsOracleTestSuite.getConnectionSupplier(defaultKvs);
+        oracleTableNameGetter = OracleTableNameGetterImpl.createDefault((OracleDdlConfig) CONFIG.ddl());
+    }
+
+    @After
+    public void tearDown() {
+        defaultKvs.dropTables(defaultKvs.getAllTableNames());
+        connectionSupplier.close();
+    }
+
+    @Test
+    public void whenConfiguredAlterTableToMatchMetadataAndOldDataIsStillReadable() {
+        defaultKvs.createTable(TABLE_REFERENCE, EXPECTED_TABLE_METADATA.persistToBytes());
+        writeData(defaultKvs);
+        assertThatDataCanBeRead(defaultKvs);
+        dropOverflowColumn(defaultKvs);
+        defaultKvs.putMetadataForTable(TABLE_REFERENCE, OLD_TABLE_METADATA.persistToBytes());
+        assertThatThrownBy(() -> fetchData(defaultKvs));
+        ConnectionManagerAwareDbKvs workingKvs = ConnectionManagerAwareDbKvs.create(CONFIG_WITH_ALTER);
+        workingKvs.createTable(TABLE_REFERENCE, EXPECTED_TABLE_METADATA.persistToBytes());
+        assertThatDataCanBeRead(workingKvs);
+        assertThatOverflowColumnExists(workingKvs);
+        assertThatDataCanBeWritten(workingKvs);
+    }
+
+    @Test
+    public void whenConfiguredAlterTableDoesNothingWhenMatching() {
```

Comment:
Is this test just trying to assert that if the config is present but the table is already fine we are still fine?

Reply:
Yup, just b/c I wanted to prove idempotency, although I'm not sure why I didn't include testing reads/writes _again_ after calling createTable twice